src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,414 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+a list of these variable names and the `myTask2` pragma an `import` with a list including the same names. Example:
+
+`myTask1` implementation bloc would be:
+```python
+#%task(name=myTask1, export=[var1,var2])
+var1 = "Hello"
+var2 = "ActiveEon!"
+```
+
+and `myTask2` implementation bloc would be:
+```python
+#%task(name=myTask2, dep=[myTask1], import[var1,var2])
+print(var1 + " from " + var2)
+```
+
+====== Implementation file
+
+It is possible to use an external implementation file as task implementation. To do so, the option `path` should be used.
+Example:
+
+```python
+#%task(name=myTask,path=PATH_TO/IMPLEMENTATION_FILE.py)
+```
+
+===== Imports libraries
+
+The main difference between Proactive kernel and a native language one resides in the memory access during the
+execution of the different blocs.  While in a common native language kernel the whole script code (all the notebook
+blocs) is executed locally in the same shared memory zone, the ProActive kernel each created task will be executed as
+an independent process.
+To facilitate the transition from native language kernels to the ProActive one we included the pragma `#%import()`.
+This pragma gives the ability to the user to add, just once in the notebook, libraries that are common to all created
+tasks implemented in a same native script language.
+
+The pragma is used in this general manner: `#%import([language=SCRIPT_LANGUAGE])`. Example:
+
+```python
+#%import(language=Python)
+import os
+import pandas
+```
+
+Notice that if the language is not specified, Python is considered by default.
+
+===== Adding a fork environment
+
+To configure a fork environment for a task, please use the `#%fork_env()` pragma. A first way to do this
+is by providing the name of the corresponding task, and the fork environment implementation after that:
+
+```text
+#%fork_env(name=TASK_NAME)
+containerName = 'activeeon/dlm3'
+dockerRunCommand =  'docker run '
+dockerParameters = '--rm '
+paHomeHost = variables.get("PA_SCHEDULER_HOME")
+paHomeContainer = variables.get("PA_SCHEDULER_HOME")
+proActiveHomeVolume = '-v '+paHomeHost +':'+paHomeContainer+' '
+workspaceHost = localspace
+workspaceContainer = localspace
+workspaceVolume = '-v '+localspace +':'+localspace+' '
+containerWorkingDirectory = '-w '+workspaceContainer+' '
+preJavaHomeCmd = dockerRunCommand + dockerParameters + proActiveHomeVolume + workspaceVolume + containerWorkingDirectory + containerName
+```
+
+A second way is by providing the name of the task, and the path of a _.py_ file containing the fork environment code:
+
+```python
+#%fork_env(name=TASK_NAME, path=PATH_TO/FORK_ENV_FILE.py)
+```
+
+===== Adding a selection script
+
+To add a selection script to a task, please use the `#%selection_script()` pragma. A first way to do it,
+provide the name of the corresponding task, and the selection code implementation after that:
```

Comment:
A first way to do it is by providing the name... (reads better like below "A second way to do it...")